ssl/t1_lib.c

```diff
@@ -3907,11 +3922,25 @@ static int tls1_check_sig_alg(SSL_CONNECTION *s, X509 *x, int default_nid)
         sigalgslen = s->shared_sigalgslen;
     }
     for (i = 0; i < sigalgslen; i++) {
+        int mdnid, pknid;
+
         sigalg = use_pc_sigalgs
                  ? tls1_lookup_sigalg(SSL_CONNECTION_GET_CTX(s),
                                       s->s3.tmp.peer_cert_sigalgs[i])
                  : s->shared_sigalgs[i];
-        if (sigalg != NULL && sig_nid == sigalg->sigandhash)
+        if (sigalg == NULL)
+            continue;
+        if (sig_nid == sigalg->sigandhash)
+            return 1;
+        if (sigalg->sig != EVP_PKEY_RSA_PSS)
+            continue;
+        /*
+         * Accept RSA PKCS#1 signatures in certificates when the signature
+         * algorithms include RSA-PSS with a matching digest algorithm.
```

Comment:
Perhaps some elaboration as to why we are accepting them.

Reply:
I'll also defer the comments just for now, but yes, they should be added.